Added a new feature that provides users to view what type is superior or inferior over the other

TheNova22 opened the pull request:
Hey! So tried to add a screen to view different types that pokemon have and using which type will be better over the other. The below are the changes/additions that i made nearly in order of the commits i made to my fork:

- Created a custom list that stored all the types and it's effects over the others. Didn't prefer a json file or use an api/save in gist as there are only 18 types and hence only 18 elements, so the list is small.

- Added a copyright free gif of pikachu running at the starter page above the "Pokedex" text. Credits for the gif given in README. (Note: When run on debug mode, the gif will appear slow, so if u want to run the app to see the gif work, use release mode)

- A bottom overflow in the pokedex page was noticed when the search button was clicked on from the FAB button. Corrected that issue.

- Added the type logos/images/icons that are also copyright free and whose credits are given in the README section.

- Then used the icons and made an appearance of enclosing them within a pokeball and displayed each of them as a grid.
(Note : The count of grid can be changed as you like in the code, but also change the width of the inner pokeball circle while changing the count, else the pokeball will end up looking like a regular ball)

- Then created a function that showed a bottom sheet with all the relevant info about the type like what types is it super effective to and what type it is not at all effective to.

- Combined all of these together and displayed it in the earlier "Type Charts" section.

Below are some of the screenshots for better understanding.

![WhatsApp Image 2020-10-04 at 1 55 54 PM](https://user-images.githubusercontent.com/51517054/95476388-d0f5fa80-09a4-11eb-9cd9-0b68800b358f.jpeg)



![WhatsApp Image 2020-10-08 at 8 10 31 PM](https://user-images.githubusercontent.com/51517054/95476401-d5221800-09a4-11eb-8bf8-9eaf42c54ff2.jpeg)



![WhatsApp Image 2020-10-08 at 8 11 07 PM](https://user-images.githubusercontent.com/51517054/95476407-d7847200-09a4-11eb-92d4-ba87488f03a1.jpeg)



![WhatsApp Image 2020-10-08 at 8 12 25 PM](https://user-images.githubusercontent.com/51517054/95476418-d9e6cc00-09a4-11eb-8aa3-67110bb2a6b7.jpeg)


Hope you like it and merge it :)
It may not to be the finest piece of ui i guess, but the code is perfect and easily editable. I tried my best in making this happen and finally succeeded too. Hence happy with what i have done so far and will continue to do so.

TheNova22 commented:
And also would you mind labelling this individual pull request as "hacktoberfest-accepted", as its something i planned to participate and this is the repo i am enjoying working on as it's Flutter and Pokemon in one! So would be glad if you could do that.  :)

scitbiz commented:
Hi @TheNova22,

Another great pull again! It's looking good. However, I have a few comments about the Type Effects screen:
- The project is focusing on creating just like everything you can with Flutter and show how powerful Flutter is, and I think all of these images can be converted to Flutter's widgets by using CustomPaint or CustomClipper. Try to make it and if you having any trouble, just let me know and I can help you with that.
- The app is following up on the original design. Since the author doesn't make the Type Effects screen, so I can understand that you need to think and create the screen by yourself. However, it will be awesome if you can create the UI with the same material as the design (like flat icon/image, same spacing/padding, use the same color of the pokemon card), that will make the app sharing the same design, not just a combination of multiple app screens with a different design. 
- If the JSON file is small, try to make the class of it so we can easily use it without guessing its key and type. There are other guys already do it in [here](https://github.com/scitbiz/flutter_pokedex/pull/14/files#diff-a9045abe8e95f0ea90c55617801bde47), I haven't rebased it to the refactor branch yet, but you can check it out and create your own version and I will do the rest when rebasing.

Once again, thanks for contributing to this repository. I really appreciate that!

scitbiz commented:
> And also would you mind labelling this individual pull request as "hacktoberfest-accepted", as its something i planned to participate and this is the repo i am enjoying working on as it's Flutter and Pokemon in one! So would be glad if you could do that. :)

About labeling `hacktoberfest-accepted`, I do not know what it is, so I cannot add it to the pull request. However, after rebasing the `refactor` branch to the `master` branch, I will add all contributors to the README to show my appreciation to them. And you are definitely one of the guys in there.

scitbiz commented on lib/ui/screens/splash/splash.dart:
```diff
@@ -24,7 +24,18 @@ class _SplashScreenState extends State<SplashScreen> {
-        child: Text('Pokedex'),
+        child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+          Image.asset(
+            "assets/gif/pika_loader.gif",
```
Define the image in `lib/configs/images.dart` and use it like:

```dart
Image(
    image: AppImages.pikaLoader,
    ...
)
```

scitbiz commented on lib/ui/screens/types/typeCard.dart:
```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex/configs/types.dart';
+import 'package:pokedex/ui/screens/types/typeSheet.dart';
```
The `typeCard.dart` and `typeSheet.dart`, etc is not a Dart suggested name format. Please change it to `type_card.dart` and `type_sheet.dart`

scitbiz commented on lib/ui/screens/types/typeCard.dart:
```diff
@@ -0,0 +1,83 @@
+class PokeTypes extends StatelessWidget {
+  const PokeTypes({
+    Key key,
+    @required this.w,
```
Please write the property name clearly

scitbiz commented on lib/ui/screens/types/typeListView.dart:
```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex/configs/types.dart';
+
+List<Widget> _lister(int i, dynamic n, double w) {
```
Please move this function into a method of `TypeFullList`

scitbiz commented on lib/ui/screens/types/type_screen.dart:
```diff
@@ -0,0 +1,76 @@
+                  ),
+                ),
+              ),
+              TypeGrid()
```
The name of `TypeCard, TypeListView, TypeGrid, TypeSheet` is not clear enough and could be used by accident in another screen/widgets. Change it to something like `TypeEffectCard, TypeEffectList, TypeEffectGrid, TypeEffectSheet` instead

TheNova22 commented:
Thanks for the reply! Will surely try and get the changes requested corrected soon. Then will try getting into the changes you mentioned in the first comment of yours and try making them happen. If faced by any issue or problem, will let you know for some guidance.

scitbiz commented:
Hi @TheNova22,
The master branch has been updated to the newest version. Please change the target branch of the pull request to `master` and rebase your fork if you still want to keep working on this. Thanks!

TheNova22 commented:
Hey @scitbiz 
Yup, i am working on this and will continue to do so. It's just that i had some other things to tend to that i couldn't make the changes yet. Will from me soon!
Glad that you have merged it with master branch :)

TheNova22 commented:
Hello, made the requested changes and also changed the UI a bit to make it look better. The below are the changes made:

- Deleted the images of the types and made the type images using code itself with the help of certain widgets.
- Couldn’t make the pokeball using Clipper as i am not so familiar with its functioning yet, but once i understand and know how it works, might change and contribute to that too.
- Changed the list of maps to class as mentioned and used the predefined enum types.
- Used the ideal Color code as requested i.e. 0xFFFFFF. Couldn’t use the colors declared in AppColors as the pokeballs didn’t seem that ideal according to me, but these color can easily be changed if new colors are added.
- Sorry about that the bad variable naming and file naming in the previous commits, made changes to them and also added comments explaining them.
- Added buttons to redirect to required pages that can be used in future when filter search is added to the app. This can be seen in the attached image below where ‘Fighting Type Pokemoms’, etc can be seen.
- As mentioned in the previous comments, the pikachu gif might be seen slow during debugging, but when seen during release, it works perfectly.
- Made the code short and as readable as possible to avoid confusions.
- Again glad that refactor now has been merged with master branch. Hoping that this PR too gets merged and helps all the pokefans out there :)

 ![WhatsApp Image 2020-11-10 at 8 15 03 PM](https://user-images.githubusercontent.com/51517054/98693438-4edb6600-2396-11eb-90ab-6448ff5d41e7.jpeg) 

 ![WhatsApp Image 2020-11-10 at 8 15 39 PM](https://user-images.githubusercontent.com/51517054/98693437-4e42cf80-2396-11eb-8028-bea371372726.jpeg)   ![WhatsApp Image 2020-11-10 at 8 16 14 PM](https://user-images.githubusercontent.com/51517054/98693432-4d11a280-2396-11eb-871e-186b6af7d234.jpeg)

scitbiz commented:
Hi @TheNova22,
It's a lot of improvements since last time. There are some few problems left, but it's okay, I will fix it later.
But there are so many commits, some of them are duplicated and unnecessary. Please squash them to be like 3-4 commits only. 

I appreciate your efforts in putting on this repo! If you have any problems, just let me know.

TheNova22 commented:
Hi @scitbiz 
Glad to know that it’s in a stage worthy to merge. But can i close this PR and start a new one instead of squashing the several commits that i carelessly did for even a small change. This is because i am unable to squash for some problem faced in my end, so will be glad if your ok with me closing this PR and then I will immediately make a new PR as to let you merge it.
Glad to contribute to such a fun project :)

scitbiz commented:
@TheNova22 Feel free to close the issue and open a new one if it's work for you.

TheNova22 commented:
Sure, Thanks for the quick reply!
